app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectCreationActivity.kt

```diff
@@ -171,7 +198,7 @@ class Form : AppCompatActivity() {
      */
     private fun constructProject(): Result<ImmutableProject> {
         return ImmutableProject.build(
-            id = "", //id is defined by firebase itself
+            id = if(projectToEdit == null) "" else projectToEdit!!.id, //id is defined by firebase itself
```

Comment:
Using projectToEdit?. { project ->} is better because it wont cause the app to crash